File: abc/abc126/abc126_c.cs
// https://atcoder.jp/contests/abc126/submissions/8399935

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var input = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
            var N = input[0];
            var K = input[1];

            var result = new double[N];
            for (int i = 0; i < N; i++)
            {
                var score = i + 1;
                result[i] = 1.0 / N;
                while (score < K)
                {
                    score *= 2;
                    result[i] *= 0.5;
                }
            }
            Console.WriteLine(result.Sum());
        }
    }
}

File: abc/abc126/abc126_c.fs
// https://atcoder.jp/contests/abc126/submissions/12208940

let [| N; K |] = stdin.ReadLine().Split() |> Array.map int

let r = Array.zeroCreate<float> N

let solver =
    let rec loop i =
        if i = N then ()
        else
            let mutable s = i + 1
            r.[i] <- 1. / (float N)
            while s < K do
                s <- s * 2
                r.[i] <- r.[i] * 0.5
            loop (i + 1)
    loop 0

solver
r
|> Array.sum
|> stdout.WriteLine
